### PR TITLE
refactor(tree): switch html structure from div to ul

### PR DESCRIPTION
**Pull request by ingeridhellen**
## What does this pull request change?

Before, the tree used divs upon divs. But actually, a nested list with buttons is a more correct approach, so I fixed that. I've also kept further improvements in mind while working with this, so I hope I easily can add new functionality on top of this.

## Why is this pull request needed?

Makes it easier to navigate in tests and increases accessibility. Hopefully, the code is easier to maintain too.

## Issues related to this change

ref #534

### CODE REVIEW

**Comment by henrikbossart** [comment]
Screenshot 2023-09-22 at 15 05 33
This is totally a non-blocker, but could we drop the data source on these? it is already implicit as its located inside the data source directory (if that makes sense (or maybe I'm totally not getting how this works))

**Review by collinlokken — APPROVED** [review]
Good job!

File: packages/dm-core/src/components/TreeView.tsx [hunk 1]
```diff
@@ -19,34 +18,28 @@ import {
...
 
-type TStyledTreeNode = {
-  level: number
-}
-
-const ExpandButton = styled.div`
-  margin: 0 3px;
-  width: 15px;
+const StyledButton = styled(Button)`
+  padding: 0 8px 0 0;
```
Comment (henrikbossart) [inline packages/dm-core/src/components/TreeView.tsx:22]:
```suggestion
  padding-right: 8px;
```

Comment (henrikbossart) [reply packages/dm-core/src/components/TreeView.tsx:22]: nvm,

**Review by henrikbossart — APPROVED** [review]
oh so pretty

**Comment by ingeridhellen** [comment]
> Screenshot 2023-09-22 at 15 05 33 This is totally a non-blocker, but could we drop the data source on these? it is already implicit as its located inside the data source directory (if that makes sense (or maybe I'm totally not getting how this works))

I agree. However, that is not my doing. The TreeView just displays the name of the node, and if that does not exist, the id. Those values are set by the Tree, so that would be a different issue.

**Pull request merged**